presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java

```diff
@@ -150,11 +171,21 @@ public QueryObjectBundle rewriteQuery(@Language("SQL") String query, ClusterType
                             createTableAsSelect.getColumnAliases(),
                             createTableAsSelect.getComment()),
                     ImmutableList.of(new DropTable(temporaryTableName, true)),
-                    clusterType);
+                    clusterType,
+                    false);
         }
         if (statement instanceof Insert) {
             Insert insert = (Insert) statement;
             QualifiedName originalTableName = insert.getTarget();
+            if (shouldReuseTable) {
+                return new QueryObjectBundle(
+                        originalTableName,
```

Comment:
What if the original table contains some other data then the data inserted by the query being verifier. How do we know what partitions of the original table to compute the checksum over?

Reply:
A good question! I have a next PR on top of this one that assembles a partition predicate and applies it to the checksum query.